doc/sphinx/src/integrators.rst

```diff
@@ -0,0 +1,94 @@
+.. _integrators:
+
+Integrators
+============
+
+Integrators contain useful properties for writing a time-integration
+scheme. They are defined in a class hierarchy. ``MultiStageDriver``
+owns a pointer to an integrator. The type of integrator depends on the
+template parameter. By default it is the base class, and you can
+assign child integrator you like, but you must cast the pointer at
+runtime to access useful features of the child classes. Alternatively
+if you inherit from ``MultiStageDriver`` and specialize to a specific
+child class, you can avoid doing the cast.
+
+StagedIntegrator
+------------------
+
+The base class is the ``StagedIntegrator``. It contains a timestep
+``dt``, a total number of integration steps, ``nstages``, a total
+number of scratch buffers required, ``nbuffers``, and lists of strings
+containing suggested names for the stages and buffers, ``buffer_name``
+and ``stage_name``. All other integrators inherit from this one.
+
+LowStorageIntegrator
+----------------------
+
+
+The ``LowStorageIntegrator`` contains integrators in the 2S form as
+described in `Ketchson (2010)`_. These integrators are of the classic
+`Shu Osher`_ form:
+
+.. math::
+
+   u^{(0)} &= u^n \\
+   u^{(i)} &= sum_{k=0}^{i-1} (\alpha_{i,k} u^{(k)} + \Delta t \beta_{i, k} F(u^{(k)})\\
+   u^{n+1} &= u^{(m)}
+
+where superscripts in parentheses mean subcycles in a Runge-Kutta
+integration and :math:`F` is the right-hand-side of ODE system. The
+difference between these low-storage methods and the classic SSPK
+methods is that the low-storage methods typically have sparse
+:math:`\alpha` and :math:`\beta` matrices, which are replaced by
+diagonal termes, named :math:`\gamma_0` and :math:`\gamma_1`
+respectively. 
+
+These methods can be generalized to support more general methods with
+the introduction of an additional :math:`\delta` term for first
+averaging the updated stage with previous stages. This form is also described in Section 3.2.3 of the `Athena++ paper`_.
+
+The full update then takes the form:
+
+.. math::
+
+   u^{(1)} &:= u^{(1)} + \delta_s u^{(0)} \\
+   u^{(0)} &:= \gamma_{s0} u^{(0)} + \gamma_{s1} u^{(1)} + \beta_{s,s-1} \Delta t F(u^{(0)})
+
+where here :math:`u^{(0)}` and :math:`u^{(1)}` are the two storage
+buffers required to compute the update for a given Runge-Kutta stage
+:math:`s`.
+
+.. _Ketchson (2010): https://doi.org/10.1016/j.jcp.2009.11.006
+
+.. _Shu Osher: https://doi.org/10.1016/0021-9991(88)90177-5
+
+.. _Athena++ paper: https://doi.org/10.3847/1538-4365/ab929b
+
+The ``LowStorageIntegrator`` contains arrays for ``delta``, ``beta``,
+``gam0``, and ``gam1``. Available integratioon methods are:
+
+* ``RK1``, which is simply forward Euler
+
+* ``RK2``, which is Heun's method
+
+* ``VL2``
+
+* ``RK3``, a strong stability preserving variant
+
+* ``RK4``, a strong stability preserving variant
+
+ButcherIntegrator
```

Comment:
What is the use case for `ButcherIntegrator`? It is really neat to see a 10th order integrator working in Parthenon, but is it practical (i.e., will people use it?).  Also, why would someone use `ButcherIntegrator` for `RK1`, `RK2`, `RK4` instead of `LowStorageIntegrator `

Reply:
Couldn't you still do this with `LowStorageIntegrator` `RK1` (coupled with donor cell reconstruction, of course)?  Why do you need the `ButcherIntegrator` `RK1`?

Reply:
I think you need the ButcherIntegrator RK2/RK3/RK4, because it keeps each stage, so you can then reconstruct the lower-order result for a small number of cells from the intermediate stages.

Reply:
I had IMEX and other multi-rate methods in mind, not all of which can be expressed in low-storage form.